### PR TITLE
push public server image to public gcr repo, fix build

### PR DESCRIPTION
**Pull request by OmerGery**

### CODE REVIEW

File: .github/workflows/autofleet_server_deploy.yml [hunk 1]
```diff
@@ -5,11 +5,11 @@ on:
     paths:
       # re-deploy if a change has been made to either the server or to the example
       - 'examples/server/**'
-      - 'server/**'
```
Comment (OfryL) [inline .github/workflows/autofleet_server_deploy.yml:8]: why?

File: examples/server/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM autofleet/locomotion:2.1.7
+FROM gcr.io/af-gcr/locomotion-server:add-push-to-docker-hub-latest
```
Comment (oriefrati) [inline examples/server/Dockerfile]: add-push-to-docker-hub??

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
 server/.env
 client/Locomotion/.env
 server/mod-germany.app.yaml
+server/dist/**
```
Comment (oriefrati) [inline .gitignore]: ignore all dist folder

File: .github/workflows/autofleet_server_deploy.yml [hunk 4]
```diff
@@ -33,7 +34,7 @@ jobs:
             path: 'examples/server'
 
   Deploy_Cloud_Run:
-    if: github.ref == 'refs/heads/master'
+    # if: github.ref == 'refs/heads/master'
```
Comment (oriefrati) [inline .github/workflows/autofleet_server_deploy.yml]: why removed?

File: .github/workflows/autofleet_server_deploy.yml [hunk 5]
```diff
@@ -4,12 +4,13 @@ on:
...
     paths:
       # re-deploy if a change has been made to either the server or to the example
-      - 'examples/server/**'
       - 'server/**'
+      - 'examples/server/**'
       - '.github/workflows/autofleet_server_deploy.yml'
 
 jobs:
   Build:
+    # if: github.ref == 'refs/heads/master'
```
Comment (oriefrati) [inline .github/workflows/autofleet_server_deploy.yml]: why removed?

Comment (OmerGery) [reply .github/workflows/autofleet_server_deploy.yml]: for debugging , i will return it

File: .github/workflows/autofleet_server_deploy.yml (hunk 4 above)
Comment (OmerGery) [reply .github/workflows/autofleet_server_deploy.yml]: for debugging , i will return it

File: examples/server/Dockerfile (hunk 2 above)
Comment (OmerGery) [reply examples/server/Dockerfile]: still for debug, will be changed to master

File: server/Dockerfile [hunk 6]
```diff
@@ -10,18 +10,22 @@ RUN echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc
...
 COPY package*.json ./
-
+COPY ./web-ui/package*.json ./web-ui/
 
 RUN npm install --only=production
 # If you are building your code for production
 # RUN npm install --only=production
 
-FROM node:16.13.1-slim
+FROM node:11.14.0
```
Comment (oriefrati) [inline server/Dockerfile]: why 11.1.14.0?
change to 14

File: server/Dockerfile [hunk 7]
```diff
@@ -10,18 +10,22 @@ RUN echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc
...
 
 COPY . .
+
+COPY ./web-ui ./web-ui
+RUN npm run install-webui
+
 RUN npm run build
 EXPOSE 8080
-CMD [ "node", "--max-http-header-size", "1048576", "--unhandled-rejections=warn", "./dist/index.js" ]
+CMD [ "npm", "start" ]
```
Comment (oriefrati) [inline server/Dockerfile:31]: change it to node command

File: server/package.json [hunk 8]
```diff
@@ -10,11 +10,12 @@
         "lint": "./node_modules/.bin/eslint --resolve-plugins-relative-to=. ./src",
         "dev": "nodemon",
         "dev:webui": "nodemon --ignore web-ui/build/ --exec \"NODE_ENV=development SKIP_PREFLIGHT_CHECK=true npm run build --prefix ./web-ui/ && node ./index.js\"",
-        "install-webui": "SKIP_PREFLIGHT_CHECK=true npm install --unsafe-perm --prefix ./web-ui/ && SKIP_PREFLIGHT_CHECK=true npm rebuild node-sass --prefix ./web-ui/ && SKIP_PREFLIGHT_CHECK=true npm run build --prefix ./web-ui/",
-        "start": "npm run init && node index.js",
+        "install-webui": " SKIP_PREFLIGHT_CHECK=true npm install --unsafe-perm --prefix ./web-ui/ && SKIP_PREFLIGHT_CHECK=true npm rebuild node-sass --prefix ./web-ui/ && SKIP_PREFLIGHT_CHECK=true npm run build --prefix ./web-ui/",
```
Comment (oriefrati) [inline server/package.json]: remove space at the beginning

File: server/package.json [hunk 9]
```diff
@@ -10,11 +10,12 @@
...
+        "start": "npm run init && node ./dist/src/index.js",
```
Comment (oriefrati) [inline server/package.json:14]: check if there is src folder in dist

File: server/tsconfig.json [hunk 10]
```diff
@@ -21,7 +23,7 @@
         }
     },
     "include": [
-        "src/**/*", "test"
+        "src/**/*","server/src/lib/**/*", "test", ""
```
Comment (oriefrati) [inline server/tsconfig.json]: remove empty item
place each of the items in new line

File: server/package.json (hunk 9 above)
Comment (OmerGery) [reply server/package.json:14]: it is in src

File: server/Dockerfile [hunk 11]
```diff
@@ -10,18 +10,22 @@ RUN echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc
...
 COPY package*.json ./
-
+COPY ./web-ui/package*.json ./web-ui/
 
 RUN npm install --only=production
 # If you are building your code for production
 # RUN npm install --only=production
 
-FROM node:16.13.1-slim
+FROM node:14.17.6
```
Comment (OfryL) [inline server/Dockerfile:19]: i think we have to use 14 for Build stage (now it's `FROM node:16.13.1`)

File: server/src/test/component_tests/api/v1/users.test.js [hunk 12]
```diff
@@ -37,8 +37,7 @@ describe('Users Endpoints', () => {
   };
 
   beforeAll(async () => {
-    expect((await request(app).get('/')).status).toBe(200);
-
+    expect((await request(app).get('/version')).status).toBe(200);
```
Comment (OfryL) [inline server/src/test/component_tests/api/v1/users.test.js:40]: ❤️

**Review by OfryL — APPROVED** [review]

**Pull request merged**